NewsApp-main/NewsApp/Scenes/News/ViewController/NewsViewController.swift

```diff
@@ -0,0 +1,238 @@
+//
+//  NewsViewController.swift
+//  NewsApp
+//
+//  Created by İsmail Palalı on 27.08.2023.
+//
+
+import UIKit
+import SnapKit
+
+// MARK: NewsViewDelegate
+protocol NewsViewDelegate: BaseViewDelegate {
+    func setTableView()
+    func setCollectionView()
+    func reloadData()
+    func goDetailScreen(_ sourceList: Source)
+    func showRetryPopup(title: String, message: String)
+    func hideIndicator()
+    func showIndicator()
+}
+
+final class NewsViewController: BaseViewController {
+
+    // MARK: Create UI items
+    private lazy var categoryCollectionView: UICollectionView = {
+        let layout = UICollectionViewFlowLayout()
+        layout.scrollDirection = .horizontal
+        let collectionView = UICollectionView(frame: .zero,
+                                              collectionViewLayout: layout)
+        collectionView.backgroundColor = .white
+        collectionView.showsHorizontalScrollIndicator = false
+        return collectionView
+    }()
+
+    private lazy var newsSourcesTableView: UITableView = {
+        let tableView = UITableView()
+        tableView.rowHeight = 120
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        return tableView
+    }()
+
+    private lazy var activityIndicator: UIActivityIndicatorView = {
+        if #available(iOS 13.0, *) {
+            return UIActivityIndicatorView(style: .large)
+        } else {
+            let indicator = UIActivityIndicatorView(style: .whiteLarge)
+            indicator.color = .gray
+            indicator.hidesWhenStopped = true
+            return indicator
+        }
+    }()
+
+    private let refreshControl = UIRefreshControl()
+
+    // MARK: ViewModel
+    private var networkService: NetworkServiceProtocol!
+    private var viewModel: NewsViewModel!
+
+    // MARK: lifeCycleInfo
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        activityIndicator.startAnimating()
+
+        networkService = NetworkService()
+        viewModel = NewsViewModel(service: networkService)
+        viewModel.view = self
+        viewModel.viewWillAppear()
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configure()
+    }
+
+    override var shouldAutorotate: Bool {
+        return true
+    }
+
+    override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
+        return [UIInterfaceOrientationMask.landscapeLeft,
+                UIInterfaceOrientationMask.landscapeRight,
+                UIInterfaceOrientationMask.portrait ];
+    }
```

Comment:
return gereksizse kullanmaya gerek yok

```suggestion
    override var shouldAutorotate: Bool { true  }

    override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
        [UIInterfaceOrientationMask.landscapeLeft,
                UIInterfaceOrientationMask.landscapeRight,
                UIInterfaceOrientationMask.portrait ];
    }
```